src/index.ts

```diff
@@ -1,31 +1,181 @@
-const REGION_IDS = ["auk", "bop", "can", "cit", "gis", "hkb", "mwt", "mbh", "nsn", "ntl", "ota", "stl", "tki", "tas", "wko", "wgn", "wtc"]
+const REGION_IDS = ["AUK", "BOP", "CAN", "CIT", "GIS", "HKB", "MBH", "MWT", "NSN", "NTL", "OTA", "STL", "TKI", "TAS", "WKO", "WGN", "WTC"];
 
 interface DistrictData {
-  color: string
-  description?: string
+  color?: string,
+  descriptions?: string
 }
 
 interface MapOptions {
   width?: number,
   height?: number,
+  targetSelector: string,
+  descriptions?: {
+    [key: string]: string;
+  },
+  colors?: {
+    [key: string]: string;
+  },
   data?: {
     [key in typeof REGION_IDS[number]]: DistrictData
   }
 }
 
 class NZMapSVG {
-  options: MapOptions
+  options: MapOptions;
+
+  // Element tags
+  wrapper: any; //<div>
+  mapImage: any; //<svg>
+  descriptionDisplay: any //<p>
 
   constructor(options: MapOptions) {
-    this.options = options
+    this.options = options;
   }
 
   // Add other methods here
+  render() {
+    if (
+      !this.options.targetSelector ||
+      !document.querySelector(`#${this.options.targetSelector}`)
```

Comment:
Just do `document.querySelector(this.options.targetSelector})`

You can pass in any css selector to `document.querySelector `

Reply:
E.g. you can pass in 
{ ...targetSelector: "#map" }
or 
{ ...targetSelector: ".container > .map" }
or 
{ ...targetSelector: "div.map" } etc.
https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector

Reply:
so this.options.targetSelector == map
`#${this.options.targetSelector}` == #map

the first one doesn't work